### PR TITLE
refactor: Kotlin migration of Text importer

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
What really matters here is that both commits are done with #10779

### CODE REVIEW

**Review by david-allison — CHANGES_REQUESTED** [review]
ClassName is incorrect. See comment above the line in kotlinMigration for the expected format

I'm on my phone and it won't let me comment on the commit

File: AnkiDroid/kotlinMigration.gradle [hunk 1]
```diff
@@ -43,7 +43,7 @@ permission notice:
 // Example of class name: "/com/ichi2/anki/UIUtils.kt"
 // Ensure that it starts with '/' (slash)
 def source = Source.MAIN
-def className = ""
+def className = "/AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt"
```
Comment (david-allison) [inline AnkiDroid/kotlinMigration.gradle]: /com/ichi2/libanki/importer/TextImporter.kt

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt [hunk 2]
```diff
@@ -1,313 +1,262 @@
...
+        mDialect = null
+        mFileobj = FileObj.open(mFile)
+        val firstLine = firstFileLine.orElse(null)
         if (firstLine != null) {
             if (firstLine.startsWith("tags:")) {
-                String tags = firstLine.substring("tags:".length()).trim();
-                mTagstoadd = tags.split(" ");
-                this.mFirstLineWasTags = true;
+                val tags = firstLine.substring("tags:".length).trim { it <= ' ' }
+                mTagstoadd = tags.split(" ".toRegex()).toTypedArray()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:117]: cleanup: doesn't need regex

File: AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt [hunk 3]
```diff
@@ -1,313 +1,262 @@
...
+        private val COMMENT_PATTERN = Pattern.compile("^#.*$", Pattern.DOTALL)
     }
 
-    /** Non-UTF file was provided (for example: an image) */
-    public static class EncodingException extends RuntimeException {
-        public EncodingException(Exception e) {
-            super(e);
-        }
+    init {
+        mTagstoadd = emptyArray()
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/importer/TextImporter.kt:260]: cleanup: inline

**Review by david-allison — CHANGES_REQUESTED** [review]
Looks good, very few comments

**Comment by mikehardy** [comment]
Appears to have taken a legitimate (non-flake) emulator test failure:

```
com.ichi2.anki.tests.libanki.ImportTest > testCsv[TestingAVD(AVD) - 11] FAILED 
	java.lang.AssertionError:
	Expected: iterable containing ["‘多すぎる too many fields’ had 3 fields, expected 2", "‘not, enough, fields’ had 1 fields, expected 2", "Appeared twice in file: 飲む", "Empty first field:  to play", "5 notes added, 0 notes updated, 0 notes unchanged."]
```

I re-ran emulator tests just in case it was a flake though.

It is marked as "waiting for feedback" not sure what the actual status is, and it will depend a little bit on emulator CI passing or not (if it fails again it's definitely "waiting for feedback", otherwise it might be "ready for review" again?

**Pull request closed without merging**